discovery/syncer.go

```diff
@@ -929,69 +946,80 @@ func (g *GossipSyncer) replyChanRangeQuery(query *lnwire.QueryChannelRange) erro
 	// TODO(roasbeef): means can't send max uint above?
 	//  * or make internal 64
 
-	// In the base case (no actual response) the first block and last block
-	// will match those of the query. In the loop below, we'll update these
-	// two variables incrementally with each chunk to properly compute the
-	// starting block for each response and the number of blocks in a
-	// response.
-	firstBlockHeight := startBlock
-	lastBlockHeight := endBlock
-
-	numChannels := int32(len(channelRange))
-	numChansSent := int32(0)
+	// We'll send our response in a streaming manner, chunk-by-chunk. We do
+	// this as there's a transport message size limit which we'll need to
+	// adhere to.
+	var channelChunk, prevChannelChunk []lnwire.ShortChannelID
 	for {
-		// We'll send our this response in a streaming manner,
-		// chunk-by-chunk. We do this as there's a transport message
-		// size limit which we'll need to adhere to.
-		var channelChunk []lnwire.ShortChannelID
-
-		// We know this is the final chunk, if the difference between
-		// the total number of channels, and the number of channels
-		// we've sent is less-than-or-equal to the chunk size.
-		isFinalChunk := (numChannels - numChansSent) <= g.cfg.chunkSize
-
-		// If this is indeed the last chunk, then we'll send the
-		// remainder of the channels.
-		if isFinalChunk {
-			channelChunk = channelRange[numChansSent:]
+		// In the base case (no actual response), the first block and
+		// last block will match those of the query.
+		firstBlockHeight := startBlock
+		lastBlockHeight := endBlock
+
+		nextIdx := 0
+		for i, channels := range channelRange {
```

Comment:
Why iterate over the entire range each time vs keeping a distinct index?

Reply:
Also possibly rename the variable to something like `chanRangeByBlock` since the response is now itself _nested_ (grouped by block height).

Reply:
Ended up reworking this whole logic. Initially, I wanted to maintain the existing structure while implementing the chunk splitting, but it ended up becoming less readable. It should be much easier to follow now.